ownCloud/UI Elements/DownloadFileProgressHUDViewController.swift

```diff
@@ -82,8 +83,9 @@ class DownloadFileProgressHUDViewController: UIViewController {
 		NSLayoutConstraint.activate([
 			progressView.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor),
 			progressView.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor),
-			progressView.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: progressViewSidesConstraintConstant),
-			progressView.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor, constant: -progressViewSidesConstraintConstant)
+			progressView.leftAnchor.constraint(greaterThanOrEqualTo: view.safeAreaLayoutGuide.leftAnchor, constant: progressViewSidesConstraintConstant),
+			progressView.rightAnchor.constraint(greaterThanOrEqualTo: view.safeAreaLayoutGuide.rightAnchor, constant: progressViewSidesConstraintConstant),
```

Comment:
This should be changed to be effective:

```suggestion
			progressView.rightAnchor.constraint(lessThanOrEqualToConstant: view.safeAreaLayoutGuide.rightAnchor, constant: -progressViewSidesConstraintConstant)
```

Reply:
I think `lessThanOrEqualTo ` is not what I need, because I need a flexible width for the button and width should not larger than 280 px. You can only get this with `greaterThanOrEqualTo ` or do I miss something?

Reply:
For the right anchor, one needs to think in the opposite direction. It becomes clearer when putting in numbers:

```
progressView.rightAnchor <= view.safeAreaLayoutGuide.rightAnchor + (-progressViewSidesConstraintConstant)
```

For `view.safeAreaLayoutGuide.rightAnchor = 320` (assuming a display width of 320 points) that'd mathematically equal (my proposed):

```
progressView.rightAnchor <= 320 + (-progressViewSidesConstraintConstant)
…
progressView.rightAnchor <= 320 - 20
…
progressView.rightAnchor <= 300
```

instead of (the previous):

```
progressView.rightAnchor >= 320 + progressViewSidesConstraintConstant
…
progressView.rightAnchor >= 320 + 20
…
progressView.rightAnchor >= 340
```

IMO the latter won't give the desired result because it'd ask Auto-Layout to extend the progress view's right side 20 pixels outside the visible area.

Reply:
Turns out when running the code, the constraints for the label and button hang on the `progressView` and cause issues as well. I ended up with this code:

```swift
	override func loadView() {
		super.loadView()

		view.backgroundColor = UIColor.init(white: rootViewBackgroundWhite, alpha: rootViewBackgroundAlpha)

		// Progress view
		progressView.translatesAutoresizingMaskIntoConstraints = false
		view.addSubview(progressView)
		NSLayoutConstraint.activate([
			progressView.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor),
			progressView.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor),
			progressView.leftAnchor.constraint(greaterThanOrEqualTo: view.safeAreaLayoutGuide.leftAnchor, constant: progressViewSidesConstraintConstant).with(priority: .defaultHigh - 100),
			progressView.rightAnchor.constraint(lessThanOrEqualTo: view.safeAreaLayoutGuide.rightAnchor, constant: -progressViewSidesConstraintConstant).with(priority: .defaultHigh - 100),
			progressView.widthAnchor.constraint(lessThanOrEqualToConstant: progressViewWidthConstraintConstant).with(priority: .defaultHigh)
		])

		// Cancel button
		cancelButton.translatesAutoresizingMaskIntoConstraints = false
		view.addSubview(cancelButton)
		NSLayoutConstraint.activate([
			cancelButton.centerXAnchor.constraint(equalTo: progressView.centerXAnchor),
			cancelButton.topAnchor.constraint(equalTo: progressView.bottomAnchor, constant: cancelButtonTopAnchor)
		])
		cancelButton.setTitle("Cancel".localized, for: .normal)

		// Info label
		infoLabel.translatesAutoresizingMaskIntoConstraints = false
		infoLabel.setContentHuggingPriority(.required, for: .vertical)
		view.addSubview(infoLabel)
		NSLayoutConstraint.activate([
			infoLabel.bottomAnchor.constraint(equalTo: progressView.topAnchor, constant: -infoLabelBottomAnchor),
			infoLabel.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor),
			infoLabel.leftAnchor.constraint(greaterThanOrEqualTo: view.safeAreaLayoutGuide.leftAnchor, constant: progressViewSidesConstraintConstant),
			infoLabel.rightAnchor.constraint(lessThanOrEqualTo: view.safeAreaLayoutGuide.rightAnchor, constant: -progressViewSidesConstraintConstant)
		])

		infoLabel.text = "Downloading".localized
		infoLabel.textColor = .white
		infoLabel.textAlignment = .center

		progressView.progressTintColor = .white
		progressView.trackTintColor = .lightGray
	}
```

Looking at `DownloadFileProgressHUDViewController`, I generally wonder if we still need it at all, since:
- we now have inline progress reporting/cancellation (`DownloadFileProgressHUDViewController` predates that)
- the user can't do anything else while a file downloads for opening

Reply:
Thank you for fixing my mistake! 
I will open a new issue, if we still need the `DownloadFileProgressHUDViewController`.